pkg/frontend/querymiddleware/spin_off_subqueries_queryable.go

```diff
@@ -110,62 +110,64 @@ func (q *spinOffSubqueriesQuerier) Select(ctx context.Context, _ bool, hints *st
 		if err != nil {
 			return storage.ErrSeriesSet(errors.Wrap(err, "failed to parse subquery step"))
 		}
+		var queryOffset time.Duration
+		if offsetStr == "" {
+			queryOffset = 0
+		} else if queryOffset, err = time.ParseDuration(offsetStr); err != nil {
+			return storage.ErrSeriesSet(errors.Wrap(err, "failed to parse subquery offset"))
+		}
 
-		end := q.req.GetEnd()
-		// Align query to absolute time by querying slightly into the future.
+		end := q.req.GetEnd() - queryOffset.Milliseconds()
```

Comment:
Might be worth adding a note here to make it clear that this would need to change if we want to support range queries in addition to instant queries.